securebanking-openbanking-uk-rs-simulator-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/obie/payment/v3_1_8/vrp/DomesticVrpsApiController.java

```diff
@@ -145,7 +156,44 @@ public ResponseEntity<OBDomesticVRPResponse> domesticVrpPost(
     ) throws OBErrorResponseException {
         log.debug("Received VRP payment submission: '{}'", obDomesticVRPRequest);
 
+        String consentId = obDomesticVRPRequest.getData().getConsentId();
+        //get the consent
+        JsonObject intent = consentService.getIDMIntent(authorization, consentId);
+
+        log.debug("Retrieved consent from IDM");
+
+        //deserialize the intent to ob response object
+        OBDomesticVRPConsentResponse consent = consentService.deserialize(
+                OBDomesticVRPConsentResponse.class,
+                intent.getAsJsonObject("OBIntentObject"),
+                consentId
+        );
+
+        log.debug("Deserialized consent from IDM");
+
+        //get mandatory objects
+        OBDomesticVRPInitiation initiation = consent.getData().getInitiation();
+        OBDomesticVRPControlParameters controlParameters = consent.getData().getControlParameters();
+        OBRisk1 risk = consent.getRisk();
+        OBCashAccountCreditor3 requestCreditorAccount = consent.getData().getInitiation().getCreditorAccount();
+
+        log.debug("Validating VRP submission");
+
         FRDomesticVrpRequest frDomesticVRPRequest = toFRDomesticVRPRequest(obDomesticVRPRequest);
+
+        // validate the consent against the instruction
+        try {
+            domesticVrpValidationService.clearErrors().validate(initiation, obDomesticVRPRequest.getData().getInstruction(), risk, frDomesticVRPRequest, controlParameters, requestCreditorAccount);
+        } catch (OBErrorException e) {
+            log.error("Got error during VRP validation: '{}'", e.getMessage());
+            throw new OBErrorResponseException(
+                    e.getObriErrorType().getHttpStatus(),
+                    OBRIErrorResponseCategory.REQUEST_FILTER,
+                    e.getOBError());
```

Comment:
Return a generic error isn't right, the OBErrorResponseException returned needs to be builded with the OBErrorException returned by the validator with the proper OBErrorType returned by the validator.